compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -769,9 +768,11 @@ abstract class GenJSCode extends plugins.PluginComponent
       val superClass =
         if (sym.isTraitOrInterface) None
         else Some(encodeClassFullNameIdent(sym.superClass))
-      val jsNativeLoadSpec =
+      val jsNativeLoadSpec = {
         if (sym.isTraitOrInterface) None
+        else if (sym.hasAnnotation(JSGlobalScopeAnnotation)) None
         else Some(jsNativeLoadSpecOf(sym))
+      }
```

Comment:
Yes, it is a binary incompatible change. That's expected since it's also a *source* incompatible change now. For example, if I have
```scala
@js.native
@JSGlobal
object Foo extends js.Object
```
and
```scala
val foo = Foo
```
When I change `Foo` to `@JSGlobalScope` the statement `val foo = Foo` becomes illegal.